### DISCUSSION TITLE
Please clarify how the new AGPL license is to be applied for prediction models

### DISCUSSION THREAD

**Discussion by saschwarz** [ultralytics/ultralytics#2127] [Q&A]
In order to stay in compliance with the new AGPL license I'm hoping for some guidance on what source code needs to be available to users of an application that incorporates a YoloV8 prediction model. Some application/model deployment scenarios:

1. Trained model is deployed on a 3rd party service (e.g. Roboflow REST API, ClearML, etc.) and users interact with API directly.
2. Trained model is deployed in a 3rd party docker container that provides a REST or other API (tensorflow-serving, etc) and users interact with API directly.
3. Trained model is deployed in a custom service accessed via a REST or other API (Django, Flask, etc) and users interact with API directly.
4. Trained model is deployed via 1, 2, or 3 above and a web server communicates with the service via the API and provides a website with which users interact.
5. Trained model is deployed via 1, 2, or 3 above and a intermediate service communicates with the prediction service via the API and the intermediate service provides an API to which a client application (web site, web app, mobile app) communicates with which users interact.
6. Mobile app/PWA with embedded model.

My _guesses_ for each scenario:

1. All weights and configuration files for the model, instructions/links for deploying on that service.
2. All weights and configuration files for the model, instructions/links/scripts to deploy the container.
3. All weights and configuration files for the model, source code for the service, instructions/links/scripts to build/deploy the service.
4. 1, 2, or 3 depending on the hosting method for the model. Source for the web server is not required to be available.
5. 1, 2, or 3 depending on the hosting method for the model. Source for the intermediate service is not required to be available. Source for the client application is not required to be available.
6. The weights, configuration files for the model, and source for the mobile app/PWA must be provided.

I'm hoping someone from Ultralytics can give some guidance.

**Comment by saschwarz** [comment]
@glenn-jocher sorry to ping you directly. I'm hoping you would be able to clarify what exactly needs to be done in these different deployment scenarios?

**Reply by glenn-jocher** [reply]
@saschwarz hello! I'm glad you're looking into the licensing details for YOLOv8. It's important to ensure compliance with the AGPL-3.0 license when using the model in various applications. Here's a general guideline for each scenario you've described:

1. **3rd Party Service Deployment**: If you're using a third-party service to host the model, you need to provide the source code of the model and any modifications you've made, along with instructions for deploying the model on that service.

2. **3rd Party Docker Container**: Similar to the first scenario, you should provide all model weights, configuration files, and deployment instructions, including how to use the Docker container.

3. **Custom Service Deployment**: You must make the source code of your custom service available, including any modifications to the model, along with deployment instructions.

4. **Web Server Interaction**: The source code for the model and any service directly interacting with it must be available. The web server code that acts as a front-end does not necessarily need to be open-sourced unless it is tightly coupled with the model's operation.

5. **Intermediate Service Interaction**: The source code for the model and the immediate service interacting with it must be open-sourced. The client application or intermediate service that does not directly interact with the model may not need to be open-sourced.

6. **Mobile App/PWA with Embedded Model**: The entire application, including the model, weights, and configuration files, must be provided as it's considered a derivative work.

Remember, the AGPL-3.0 license requires that any modifications and the source code of the AGPL-licensed software (in this case, YOLOv8) be made available to users who interact with it over a network. If you're looking for more flexibility for commercial product development without the open-source requirements, you might consider obtaining an Enterprise License.

For more detailed information, please refer to our documentation at https://docs.ultralytics.com. If you have specific legal questions regarding the license, it's always best to consult with a legal professional.
